client/src/VerificationController.ts

```diff
@@ -734,17 +742,12 @@ export class VerificationController {
                 case VerificationState.VerificationRunning:
                     State.abortButton.show();
                     State.statusBarProgress.show();
-                    if (params.progress) {
+                    if (params.progress > 0) {
                         this.progressLabel = `Verification of ${params.filename}:`;
                         this.addTiming(params.filename, params.progress, Color.ACTIVE);
                     }
                     if (params.diagnostics) {
-                        let diagnostics: vscode.Diagnostic[] = [];                        
-                        JSON.parse(params.diagnostics).forEach( item => {
-                            let range = new vscode.Range(item.range.start.line, item.range.start.character, item.range.end.line, item.range.end.character);
-                            let diag = new vscode.Diagnostic(range, item.message, item.severity-1);
```

Comment:
I suspect that the yellow squiggly line issue comes from the fact that the severity level is not set properly. 
If you want to use only one severity level, at least I would recommend to use the one that corresponds to "red" (i.e. error but not warning).